File: backend/src/main/java/com/MRSISA2021_T15/service/LoyaltyProgramService.java
package com.MRSISA2021_T15.service;

import org.springframework.http.ResponseEntity;

import com.MRSISA2021_T15.model.AppointmentConsultationPoints;
import com.MRSISA2021_T15.model.Category;
import com.MRSISA2021_T15.model.CategoryName;

public interface LoyaltyProgramService {
	
	ResponseEntity<String> defineCategories(Category category);
		
	CategoryName[] getCategoryNames();
	
	ResponseEntity<String> definePointsForAppointmentAndConsulation(AppointmentConsultationPoints appointmentConsultationPoints);
}

File: frontend/src/app/loyalty-program/loyalty-program.service.ts
import { HttpClient, HttpErrorResponse } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { throwError } from 'rxjs';
import { catchError } from 'rxjs/operators';

@Injectable({
  providedIn: 'root'
})
export class LoyaltyProgramService {

  constructor(private http: HttpClient) { }

  defineCategories(category: any) {
    return this.http.post<any>("http://localhost:8080/loyaltyProgram/defineCategories", category)
                              .pipe(catchError(this.errorHander));
  }

  getCategoryNames() {
    return this.http.get<any>("http://localhost:8080/loyaltyProgram/getCategoryNames");
  }

  definePointsForAppointmentAndConsulation(acp: any) {
    return this.http.post<any>("http://localhost:8080/loyaltyProgram/definePointsForAppointmentAndConsulation", acp)
                              .pipe(catchError(this.errorHander));
  }

  errorHander(error: HttpErrorResponse) {
    return throwError(error);
  }
}
